specification/workloadmonitor/resource-manager/Microsoft.WorkloadMonitor/preview/2020-01-13-preview/Microsoft.WorkloadMonitor.json

```diff
@@ -0,0 +1,805 @@
+      "required": true,
+      "type": "string",
+      "description": "The resourceNamespace of the resource",
+      "x-ms-parameter-location": "method"
+    },
+    "resourceType": {
+      "in": "path",
+      "name": "resourceType",
+      "required": true,
+      "type": "string",
+      "description": "The resourceType of the resource",
+      "x-ms-parameter-location": "method"
+    },
+    "resourceName": {
+      "in": "path",
+      "name": "resourceName",
+      "required": true,
+      "type": "string",
+      "description": "The resourceType of the resource",
+      "x-ms-parameter-location": "method"
+    },
+    "monitorId": {
+      "in": "path",
+      "name": "monitorId",
+      "description": "The monitorId of the resource (url encoded)",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "timestampUnix": {
+      "in": "path",
+      "name": "timestampUnix",
+      "description": "The timestamp of the state change (Unix format)",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "filter": {
+      "name": "$filter",
+      "in": "query",
+      "description": "list example: $filter=monitorName eq 'logical-disks|C:|disk-free-space-mb'; history example: $filter=isHeartbeat eq false",
+      "required": false,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "expand": {
+      "name": "$expand",
+      "in": "query",
+      "description": "ex: $expand=evidence,configuration",
+      "required": false,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "definitions": {
+    "MonitorStateChangeList": {
+      "title": "Monitor History",
+      "description": "The monitor history of a monitor",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "value": {
+          "title": "State Change Array",
+          "description": "Array of state change transitions.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MonitorStateChange"
+          }
+        },
+        "nextLink": {
+          "title": "Pageable next link",
+          "description": "Link to next page if list is too long.",
+          "type": "string"
+        }
+      }
+    },
+    "MonitorList": {
+      "title": "Monitor List",
+      "description": "Basic information about the current status of a monitor.",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "value": {
+          "title": "Monitor Array",
+          "description": "Array of monitors.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Monitor"
+          }
+        },
+        "nextLink": {
+          "title": "Pageable next link",
+          "description": "Link to next page if list is too long.",
+          "type": "string"
+        }
+      }
+    },
+    "Monitor": {
```

Comment:
For tracked resources, it is important that this uses 'allOf' to extend the tyoe for an ARM resources.  You can use a pattern like this: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json#L1349

Or you can refer to the appropriate resource in common-types here: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json#L45

Reply:
Added an allOf encompassing the underlying MonitorResource - is this correct?